relay-profiling/src/sample.rs

```diff
@@ -95,6 +126,110 @@ impl Profile {
             frame.strip_pointer_authentication_code(addr);
         }
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        let mut active_ranges: HashMap<u64, Range<usize>> = HashMap::new();
+
+        for (i, sample) in self.samples.iter().enumerate() {
+            let is_active = match self.stacks.get(sample.stack_id) {
+                Some(stack) => !stack.is_empty(),
+                None => true,
+            };
+
+            if !is_active {
+                continue;
+            }
```

Comment:
IIUC this skips the sample when there's an empty stack. Could we simplify it to something like the following?

```rs
if self.stacks.get(sample.stack_id).unwrap_or(Vec::new()).is_empty() {
  continue;
}
```
(haven't tested if it compiles)

Reply:
I changed it to something similar